src/SwedbankPay.Sdk.Tests/PaymentOrderTests.cs

```diff
@@ -1,151 +1,150 @@
-            Assert.Equal(HttpStatusCode.NotFound, thrownException.HttpResponse.StatusCode);
-        }
-    }
+﻿using SwedbankPay.Sdk.Common;
+using SwedbankPay.Sdk.Exceptions;
+using SwedbankPay.Sdk.PaymentOrders;
+using SwedbankPay.Sdk.Tests.TestBuilders;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace SwedbankPay.Sdk.Tests
+{
+    public class PaymentOrderTests : ResourceTestsBase
+    {
+        private readonly PaymentOrderRequestBuilder paymentOrderRequestBuilder = new PaymentOrderRequestBuilder();
+
+
+        [Fact]
+        public async Task CreateAndAbortPaymentOrder_ShouldReturnAbortedState()
+        {
+            //ARRANGE
+
+            var paymentOrderRequest = this.paymentOrderRequestBuilder.WithTestValues(this.payeeId).Build();
+
+            //ACT
+            var paymentOrder = await this.Sut.PaymentOrders.Create(paymentOrderRequest, PaymentOrderExpand.All);
+            Assert.NotNull(paymentOrder);
+            Assert.NotEmpty(paymentOrder.Operations);
+            Assert.NotNull(paymentOrder.Operations.Abort);
+
+            var responseContainer = await paymentOrder.Operations.Abort();
+
+            Assert.NotNull(responseContainer);
+            Assert.NotNull(responseContainer.PaymentOrder);
+            Assert.Equal(State.Aborted, responseContainer.PaymentOrder.State);
+        }
+
+
+
+        [Fact]
+        public async Task CreateAndGetPaymentOrder_ShouldReturnPaymentOrderWithSameAmountAndMetadata()
+        {
+            var paymentOrderRequest = this.paymentOrderRequestBuilder.WithTestValues(this.payeeId).Build();
+            var paymentOrder = await this.Sut.PaymentOrders.Create(paymentOrderRequest, PaymentOrderExpand.All);
+            Assert.NotNull(paymentOrder);
+            Assert.NotNull(paymentOrder.PaymentOrder);
+            var amount = paymentOrder.PaymentOrder.Amount;
+
+            var paymentOrder2 = await this.Sut.PaymentOrders.Get(paymentOrder.PaymentOrder.Id, PaymentOrderExpand.All);
+            Assert.NotNull(paymentOrder2);
+            Assert.NotNull(paymentOrder2.PaymentOrder);
+            Assert.Equal(amount.InLowestMonetaryUnit, paymentOrder2.PaymentOrder.Amount.InLowestMonetaryUnit);
+
+            Assert.Equal(paymentOrderRequest.PaymentOrder.Metadata["key1"], paymentOrder2.PaymentOrder.Metadata["key1"]);
+        }
+
+
+        [Fact]
+        public async Task CreateAndUpdateOnlyAmountOnPaymentOrder_ShouldThrowHttpResponseException()
+        {
+            var paymentOrderRequest = this.paymentOrderRequestBuilder.WithTestValues(this.payeeId).WithAmounts().Build();
+            var paymentOrder = await this.Sut.PaymentOrders.Create(paymentOrderRequest, PaymentOrderExpand.All);
+            Assert.NotNull(paymentOrder);
+            Assert.NotNull(paymentOrder.PaymentOrder);
+            var amount = paymentOrder.PaymentOrder.Amount;

```

Comment:
Codacy found an issue: [Remove the unused local variable 'amount'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)